src/rust_system/src/lib.rs

```diff
@@ -350,11 +358,30 @@ pub extern "C" fn render_depth(ptr: *mut RtDepthCamera, scene: *mut RtScene, run
     scene.rec.log("depth_cloud", &depth_image);
 
     let elapsed = start_time.elapsed();
-    println!("Render time: {:.2}ms", elapsed.as_secs_f64() * 1000.0);
-    //println!("{:?}", res.iter().fold(0.0, |acc, x| x.w.max(acc))
+    println!("Render time for rerun: {:.2}ms", elapsed.as_secs_f64() * 1000.0);*/
+    let mut boxed_data = converted_data.into_boxed_slice();
+    let data_ptr = boxed_data.as_mut_ptr();
+    let data_len = boxed_data.len();
+    std::mem::forget(boxed_data); // Prevent Rust from deallocating the memory
+
+    let elapsed2 = start_time.elapsed();
+    println!("Render time for transport: {:.2}ms", elapsed2.as_secs_f64() * 1000.0);
+
+    // Return the image data struct
+    ImageData { ptr: data_ptr, len: data_len, width, height }
     
 }
 
+#[no_mangle]
+pub extern "C" fn free_image_data(ptr: *mut u16, len: usize) {
```

Comment:
Knit: Lets try to keep the API somewhat consistent by using `free_image_data(image_data: ImageData)`